### ISSUE TITLE
Style/InverseMethods: class-hierarchy guard misses comparisons whose operands are not literal constants

### ISSUE THREAD

**Issue by corsonknowles** [rubocop/rubocop#15577]
## Expected behavior

`Style/InverseMethods` already declines to invert class-hierarchy comparisons, and says so in its documentation and in the guard's own comment:

```ruby
#   !(foo.class < Numeric) # Checking class hierarchy is allowed
```

```ruby
# When comparing classes, `!(Integer < Numeric)` is not the same as
# `Integer > Numeric`.
def possible_class_hierarchy_check?(lhs, rhs, method)
  CLASS_COMPARISON_METHODS.include?(method) &&
    (camel_case_constant?(lhs) || (rhs.size == 1 && camel_case_constant?(rhs.first)))
end
```

I would expect that protection to apply whenever the comparison is between modules, not only when an operand is written as a literal constant.

## Actual behavior

`camel_case_constant?` requires `node.const_type?`, so the guard only fires when an operand is a literal CamelCase constant. When the same comparison is written with variables — the usual shape in a helper that takes the module as an argument — the cop flags it and autocorrects to code that behaves differently.

`Module#<=` is a **partial** order: it returns `nil`, not `false`, for unrelated modules. So `!(a <= b)` and `a > b` are not equivalent.

## Steps to reproduce

`inverse_repro.rb`:

```ruby
# frozen_string_literal: true
def direct_includer?(model, included_module)
  model <= included_module && !(model.superclass <= included_module)
end
```

```console
$ rubocop --only Style/InverseMethods inverse_repro.rb
inverse_repro.rb:3:31: C: [Correctable] Style/InverseMethods: Use > instead of inverting <=.
  model <= included_module && !(model.superclass <= included_module)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

$ rubocop --only Style/InverseMethods -A inverse_repro.rb
```

Result: `model <= included_module && model.superclass > included_module`

Which inverts the meaning for the unrelated case:

```ruby
module M; end
class B; include M; end

!(Object <= M)   # => true     (Object does not include M)
Object > M       # => nil      falsy

B <= M && !(B.superclass <= M)   # => true    original
B <= M && B.superclass > M       # => nil     after autocorrect
```

`B.superclass` is `Object`, which is unrelated to `M` — the common case for this kind of "includes it directly, not via its superclass" check. So the predicate stops matching.

## Why the existing guard does not catch it

Neither operand is a `const` node: the receiver is `(send (lvar :model) :superclass)` and the argument is `(lvar :included_module)`. `possible_class_hierarchy_check?` therefore returns false, even though this is exactly the comparison its comment describes.

## Suggested fix

Broaden the guard rather than the safety note. Some options, roughly in order of how targeted they are:

1. Also treat the comparison as a hierarchy check when the receiver is a call to a method that returns a `Module` — `class`, `superclass`, `singleton_class` cover most real code, and `.class` is already the shape used in the cop's own documentation example.
2. Decline to invert `<=`, `>=`, `<`, `>` whenever neither operand is provably numeric. This is the conservative reading: for a partial order the inversion is never valid, and static analysis cannot establish totality.

I appreciate the cop is documented unsafe overall, so `-A` is required to apply this. Filing it because the cop already has a dedicated guard for precisely this hazard, and extending it would close a shape that appears whenever the module is passed in rather than named inline.

Happy to open a PR for option 1 if that direction is agreeable.

## RuboCop version

```console
$ rubocop --version
1.89.0
$ ruby -v
ruby 3.4.10 (2026-06-30 revision 2b0b7728dc) +PRISM [arm64-darwin25]
```
